src/ProjectService.Business/Commands/Task/FindTasksCommand.cs

```diff
@@ -113,37 +103,24 @@ private IGetUsersDataResponse GetUsersData(List<Guid> userId, List<string> error
       _userRepository = userRepository;
       _accessValidator = accessValidator;
       _httpContextAccessor = httpContextAccessor;
-      _findRequestValidator = findRequestValidator;
-      _rcGetCompanyEmployee = rcGetCompanyEmployee;
+      _cache = cache;
     }
 
-    public FindResultResponse<TaskInfo> Execute(FindTasksFilter filter)
+    public async Task<FindResultResponse<TaskInfo>> Execute(FindTasksFilter filter)
     {
-      FindResultResponse<TaskInfo> response = new();
-      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
-      List<DbProjectUser> projectUsers = _userRepository.Find(userId).ToList();
-
-      if (!_accessValidator.IsAdmin()
-        && !_accessValidator.HasRights(Rights.AddEditRemoveProjects)
-        && !projectUsers.Any()
-        && GetDepartment(userId, response.Errors)?.DirectorUserId != userId)
+      if (filter == null)
       {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
-
-        response.Status = OperationResultStatusType.Failed;
-        response.Errors.Add("Not enough rights.");
-
-        return response;
+        throw new ArgumentNullException(nameof(filter));
       }
 
-      if (_findRequestValidator.ValidateCustom(filter, out List<string> errors))
-      {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+      List<string> errors = new();
 
-        response.Status = OperationResultStatusType.Failed;
-        response.Errors.AddRange(errors);
+      var userId = _httpContextAccessor.HttpContext.GetUserId();
+      var projectUsers = _userRepository.Find(userId).ToList();
 
-        return response;
+      if (!(projectUsers.Any() || _accessValidator.IsAdmin()))
+      {
+        return new FindResultResponse<TaskInfo>();
```

Comment:
just new()

Reply:
mistake in merge, return code from prev commit